src/driver/franka_plan_runner.cc

```diff
@@ -442,12 +442,12 @@ bool FrankaPlanRunner::RecoverFromControlException() {
       } else {
         dexai::log()->warn(
             "RecoverFromControlException: turning safety off...");
-        SetCollisionBehaviorSafetyOff();
-        dexai::log()->warn("RecoverFromControlException: set safety off.");
-        dexai::log()->warn(
-            "RecoverFromControlException: running Franka's automatic error "
-            "recovery...");
         try {
+          SetCollisionBehaviorSafetyOff();
+          dexai::log()->warn("RecoverFromControlException: set safety off.");
+          dexai::log()->warn(
+              "RecoverFromControlException: running Franka's automatic error "
+              "recovery...");
           robot_->automaticErrorRecovery();
         } catch (const franka::Exception& e) {
```

Comment:
failure to change collision behaviour shouldn't block the automaticErrorRecovery call, should it?

Reply:
Yes it should - both setting the collision behavior and the automatic error recovery can only happen in certain modes, both fail if the robot is user stopped and ustop must be cleared first

Reply:
FYI current SetCollisionBehaviorSafetyOff and SetCollisionBehaviorSafetyOn don't make sense, as the constants are nonsensical. E.g. kMediumTorqueThreshold has some values higher than kHighTorqueThreshold.

Reply:
franka panda's specs don't even go that high
https://frankaemika.github.io/docs/control_parameters.html#limits-for-panda